Libraries/pbxspec/Sources/PBX/ProductType.cpp

```diff
@@ -195,6 +219,61 @@ parse(Context *context, plist::Dictionary const *dict, std::unordered_set<std::s
         _requiredBuiltProductsDir = RBPD->value();
     }
 
+    if (ABPs != nullptr) {
+        _allowedBuildPhases = std::vector<std::string>();
+        for (size_t n = 0; n < ABPs->count(); n++) {
+            if (auto ABP = ABPs->value<plist::String>(n)) {
+                _allowedBuildPhases->push_back(ABP->value());
+            }
+        }
+    }
+
+    if (AFTs != nullptr) {
+        _allowedFileTypes = std::unordered_map<std::string, std::vector<std::string>>();
+        for (size_t n = 0; n < AFTs->count(); n++) {
+            if (auto AFT = AFTs->value<plist::Array>(n)) {
+                auto result = _allowedFileTypes->insert({ AFTs->key(n), std::vector<std::string>() });
+
+                for (size_t m = 0; m < AFT->count(); m++) {
```

Comment:
Looks verbose; can you just construct the vector first, then insert everything after the secondary loop?

At the very least, move the vector* fetching outside of the secondary loop.